maths/ugly_numbers.py

```diff
@@ -0,0 +1,49 @@
+"""
+Ugly numbers are numbers whose only prime factors are 2, 3 or 5. The sequence
+1, 2, 3, 4, 5, 6, 8, 9, 10, 12, 15, … shows the first 11 ugly numbers. By convention,
+1 is included.
+Given an integer n, we have to find the nth ugly number.
+
+For more details, refer this article
+https://www.geeksforgeeks.org/ugly-numbers/
+"""
+
+
+def ugly_numbers(n: int) -> int:
+    """
+    Returns the nth ugly number.
+    >>> ugly_numbers(100)
+    1536
+    >>> ugly_numbers(1)
+    1
+    >>> ugly_numbers(20)
+    36
+    """
+    ugly_nums = [0] * (n)

```

Comment:
It is a Python antipattern to create and initialize a list before filling it out.  Is there a way to grow this array as the algorithm progresses.  If so, is that faster or slower than the current method?

Reply:
I dosen't have an impact on time. But I can change it like
```Python
ugly_nums = [1]
...
...
ugly_nums.append(next_num)
```

Reply:
I made this change.